citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -141,6 +141,22 @@ public ICitation getCitation(IUser user, String groupId, String key)
         }
         return attachments;
     }
+    
+    @Override
+    public List<ICitation> getNotes(IUser user, String groupId, String key)
+            throws GroupDoesNotExistException, CannotFindCitationException, ZoteroHttpStatusException {
+        ICitationGroup group = groupManager.getGroup(user, groupId);
+        if (group != null && group.getGroupId() == new Long(groupId)) {
+            if (!group.getUsers().contains(user.getUsername())) {
+                throw new AccessForbiddenException("User does not have access this citation.");
+            }
+        }
+        List<ICitation> notes = citationStore.getNotes(key);
+        if (notes.isEmpty()) {
```

Comment:
If notes have already been downloaded, are they updated at any point?

Reply:
They should get updated along while syncing the group items